### PR TITLE
Generate and Resend history v2 replication task via CLI

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
Generate and Resend history v2 replication task via CLI

**Why?**
Support operation of NDC workflow

**How did you test it?**
local test

**Potential risks**
This CLI is already broken for ndc workflow

### CODE REVIEW

File: service/history/replicatorQueueProcessor.go [hunk 1]
```diff
@@ -323,6 +323,70 @@ func GenerateReplicationTask(
 	return ret, newRunID, nil
 }
 
+// GenerateHistoryV2ReplicationTask generate replication task with history v2 replication tasks
```
Comment (meiliang86) [inline service/history/replicatorQueueProcessor.go:326]: (In the long term) We should move these into a separate replication package?

Comment (yux0) [reply service/history/replicatorQueueProcessor.go]: Yes. I think we want to keep the replication config for Kafka or RPC in the long term. So I guess a refactor is needed for this file before I move it to replication package

File: tools/cli/adminKafkaCommands.go [hunk 2]
```diff
@@ -23,11 +23,15 @@ package cli
 import (
 	"bufio"
 	"bytes"
+	ctx "context"
```
Comment (yycptt) [inline tools/cli/adminKafkaCommands.go:26]: nit: Try not to use the `ctx` alias

File: tools/cli/adminKafkaCommands.go [hunk 3]
```diff
@@ -561,7 +600,10 @@ func doRereplicate(shardID int, domainID, wid, rid string, minID, maxID int64, t
 			taskTemplate.Version = firstEvent.GetVersion()
 			taskTemplate.FirstEventID = firstEvent.GetEventId()
 			taskTemplate.NextEventID = lastEvent.GetEventId() + 1
-			task, _, err := history.GenerateReplicationTask(targets, taskTemplate, historyV2Mgr, nil, batch, common.IntPtr(shardID))
+			var err error
+			var task *replicator.ReplicationTask
+			task, _, err = history.GenerateReplicationTask(targets, taskTemplate, historyV2Mgr, nil, batch, common.IntPtr(shardID))
```
Comment (yycptt) [inline tools/cli/adminKafkaCommands.go:605]: revert the change?

File: tools/cli/adminKafkaCommands.go [hunk 4]
```diff
@@ -594,8 +636,23 @@ func AdminRereplicate(c *cli.Context) {
...
 	targets := []string{target}
 
-	producer := newKafkaProducer(c)
 	session := connectToCassandra(c)
+	adminClient := cFactory.ServerAdminClient(c)
+	var startVersion *int64
+	var producer messaging.Producer
+	if c.IsSet(FlagStartEventVersion) {
+		version := c.Int64(FlagStartEventVersion)
+		startVersion = &version
```
Comment (yycptt) [inline tools/cli/adminKafkaCommands.go:645]: `startVersion = common.Int64Ptr(c.Int64(FlagStartEventVersion))`? I think it will be easier to read.

Took me sometime to understand why you created another local `version` variable here.

File: service/frontend/adminHandler.go [hunk 5]
```diff
@@ -28,6 +28,8 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/uber/cadence/common/xdc"
```
[the remark this answers is not in the document]
Comment (yux0) [reply service/frontend/adminHandler.go:31]: done

File: tools/cli/adminKafkaCommands.go [hunk 6]
```diff
@@ -23,11 +23,15 @@ package cli
...
 	"bufio"
 	"bytes"
+	ctx "context"
 	"crypto/tls"
 	"encoding/json"
 	"errors"
 	"fmt"
 
+	"github.com/uber/cadence/.gen/go/admin"
+	serverAdmin "github.com/uber/cadence/.gen/go/admin/adminserviceclient"
```
[the remark this answers is not in the document]
Comment (yux0) [reply tools/cli/adminKafkaCommands.go:33]: done

File: tools/cli/adminKafkaCommands.go (hunk 2 above)
Comment (yux0) [reply tools/cli/adminKafkaCommands.go:26]: done

File: tools/cli/adminKafkaCommands.go (hunk 3 above)
Comment (yux0) [reply tools/cli/adminKafkaCommands.go:605]: done

File: tools/cli/adminKafkaCommands.go (hunk 4 above)
Comment (yux0) [reply tools/cli/adminKafkaCommands.go:645]: Yes. Updated
